skynet/modules/ttt/openai_api/app.py

```diff
@@ -1,69 +1,85 @@
-        init_app_state(engine_client, model_config, app.state, args)
+def initialize():
+    if not use_vllm:
+        return
 
-
-def initialize(app: FastAPI | None = None):
     log.info('Starting OpenAI API server...')
 
-    if use_vllm:
-        from vllm.entrypoints.openai.cli_args import make_arg_parser
-        from vllm.utils import FlexibleArgumentParser
-
-        parser = FlexibleArgumentParser(description="vLLM OpenAI-Compatible RESTful API server.")
-        parser = make_arg_parser(parser)
-        args = parser.parse_args(
-            [
-                '--disable-frontend-multiprocessing',  # disable running the engine in a separate process
-                '--disable-log-requests',
-                '--model',
-                llama_path,
-                '--gpu_memory_utilization',
-                '0.99',
-                '--max-model-len',
-                str(llama_n_ctx),
-                '--port',
-                str(app_port),
-            ]
-        )
-
-        asyncio.create_task(run_vllm_server(args, app))
+    proc = subprocess.Popen(
+        f'python -m vllm.entrypoints.openai.api_server \
+            --disable-log-requests \
+            --model {llama_path} \
+            --gpu_memory_utilization 0.99 \
+            --max-model-len {llama_n_ctx} \
+            --port {openai_api_server_port}'.split(),
+        shell=False,
+    )
+
+    if proc.poll() is not None:
+        log.error('Failed to start vLLM OpenAI API server')
+    else:
+        log.info('vLLM OpenAI API server started')
 
 
 async def is_ready():
     url = f'{openai_api_base_url}/health' if use_vllm else openai_api_base_url
 
     try:
-        response = await http_client.get(
-            url,
-            'text',
-            headers={'X-Skynet-UUID': app_uuid},
-        )
+        response = await http_client.get(url, 'text')
 
         if use_vllm:
             return response == ''
         else:
             return response == 'Ollama is running'
-    except Exception as e:
-        log.warning('Error checking if the server is ready: ', e)
+    except Exception:
         return False
 
 
-__all__ = ['initialize', 'is_ready']
+app = create_app()
+app.include_router(vllm_router, dependencies=dependencies, responses=responses)
+
+whitelisted_routes = ['/openai/docs', '/openai/openapi.json']
+
+bearer = JWTBearer()
+
+
+@app.middleware('http')
+async def proxy_middleware(request: Request, call_next):
+    if request.url.path in whitelisted_routes:
+        return await call_next(request)
+
+    if not bypass_auth:
+        try:
+            await bearer.__call__(request)
+        except HTTPException as e:
+            return JSONResponse(content=responses.get(e.status_code), status_code=e.status_code)
+
+    try:
+        url = f'{openai_api_base_url}{request.url.path.replace("/openai", "")}'
+        response = await http_client.request(request.method, url, headers=request.headers, data=await request.body())
```

Comment:
Doesn't this break streaming since we are awaiting for the whole body?

Reply:
is there a case where you wanna stream the request?

Reply:
🤦  True, I guess no, so we're good!